### DISCUSSION TITLE
Can the Cloudwatch Sink Modify Retention Periods?

### DISCUSSION THREAD

**Discussion by bartagas** [vectordotdev/vector#21195] [General]
Hello, first time caller here. 

I've set up Vector to push logs to Cloudwatch and have it create the log group if it doesn't exist already. I've noticed that Vector can set the retention period when it creates the log group, but won't modify it if the log group already exists.

Is this expected? I've given the proper IAM policies to 'DescribeLogGroup' and 'PutRetentionPolicy' with no success. The AWS console allows you to modify this setting anytime so I would assume Vector should be able to as well?

**Comment by jszwedko** [comment]
Hey! Yeah, that configuration is only applied when creating a log group. It could be applied to existing log groups too, but I could see the argument that that goes beyond the scope of Vector's responsibilities 🤔

**Reply by bartagas** [reply]
That is fair. I suppose I wanted to make sure I wasn't missing anything in the configuration. I also couldn't find much online which makes me think I'm the only one who wants this feature haha.

Anyway, thanks for the quick response. Appreciate it!
